pkg/yurtctl/util/kubernetes/util.go

```diff
@@ -210,6 +218,145 @@ func CreateServiceFromYaml(cliSet *kubernetes.Clientset, svcTmpl string) error {
+func CreateSecretFromYaml(cliSet *kubernetes.Clientset, ns, saTmpl string) error {
+	obj, err := YamlToObject([]byte(saTmpl))
+	if err != nil {
+		return err
+	}
+	sa, ok := obj.(*corev1.Secret)
+	if !ok {
+		return fmt.Errorf("fail to assert secret: %v", err)
+	}
+	_, err = cliSet.CoreV1().Secrets(ns).Create(context.Background(), sa, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("fail to create the secret/%s: %v", sa.Name, err)
+	}
+	klog.V(4).Infof("secret/%s is created", sa.Name)
+
+	return nil
+}
+
+// CreateMutatingWebhookConfigurationFromYaml creates the Service from the yaml template.
+func CreateMutatingWebhookConfigurationFromYaml(cliSet *kubernetes.Clientset, svcTmpl string) error {
+	obj, err := YamlToObject([]byte(svcTmpl))
+	if err != nil {
+		return err
+	}
+	svc, ok := obj.(*v1beta1.MutatingWebhookConfiguration)
+	if !ok {
+		return fmt.Errorf("fail to assert mutatingwebhookconfiguration: %v", err)
+	}
+	_, err = cliSet.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Create(context.Background(), svc, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("fail to create the mutatingwebhookconfiguration/%s: %s", svc.Name, err)
+	}
+	klog.V(4).Infof("mutatingwebhookconfiguration/%s is created", svc.Name)
+
+	return nil
+}
+
+// CreateValidatingWebhookConfigurationFromYaml creates the Service from the yaml template.
+func CreateValidatingWebhookConfigurationFromYaml(cliSet *kubernetes.Clientset, svcTmpl string) error {
+	obj, err := YamlToObject([]byte(svcTmpl))
+	if err != nil {
+		return err
+	}
+	svc, ok := obj.(*v1beta1.ValidatingWebhookConfiguration)
+	if !ok {
+		return fmt.Errorf("fail to assert validatingwebhookconfiguration: %v", err)
+	}
+	_, err = cliSet.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Create(context.Background(), svc, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("fail to create the validatingwebhookconfiguration/%s: %s", svc.Name, err)
+	}
+	klog.V(4).Infof("validatingwebhookconfiguration/%s is created", svc.Name)
+
+	return nil
+}
+
+func CreateCRDFromYaml(clientset *kubernetes.Clientset, yurtAppManagerClient dynamic.Interface, nameSpace string,filebytes []byte) error {
+	var err error
+	decoder := yamlutil.NewYAMLOrJSONDecoder(bytes.NewReader(filebytes), 10000)
+	for {
+		var rawObj runtime.RawExtension
+		if err = decoder.Decode(&rawObj); err != nil {
+			break
+		}
+		obj, gvk, err := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme).Decode(rawObj.Raw, nil, nil)
+		unstructuredMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+		if err != nil {
+			return err
+		}
+		unstructuredObj := &unstructured.Unstructured{Object: unstructuredMap}
+		gr, err := restmapper.GetAPIGroupResources(clientset.Discovery())
+		if err != nil {
+			return err
+		}
+
+		mapper := restmapper.NewDiscoveryRESTMapper(gr)
+		mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
```

Comment:
When was the gvk information of crd registered?

Reply:
Before creating CRD.